src/routes/dashboard/streams/[streamId=i32]/topics/[topicId=i32]/partitions/[partitionId=i32]/messages/+page.svelte

```diff
@@ -0,0 +1,73 @@
+<script lang="ts">
+  import { page } from '$app/stores';
+  import Button from '$lib/components/Button.svelte';
+  import Icon from '$lib/components/Icon.svelte';
+  import { goto } from '$app/navigation';
+  import { openModal } from '$lib/components/Modals/AppModals.svelte';
+  import SortableList from '$lib/components/SortableList.svelte';
+  import Paginator from '$lib/components/Paginator.svelte';
+
+  export let data;
+  $: topic = data.topic;
+  $: partitionMessages = data.partitionMessages;
+  $: prevPage = $page.url.pathname.split('/').slice(0, 6).join('/') + '/';
+
+  let currentPage = 1;
+  $: totalPages = Math.ceil(partitionMessages.currentOffset / data.pagination.count);
+
+  async function loadPage(page: number) {
+    const offset = (page - 1) * data.pagination.count;
```

Comment:
What is the order in which messages are paginated ? from lowest to highest ? if so I think it's better idea to display newest messages (largest offset) on first pages, I think `const offset = (totalPageNumber - page) * data.pagination.count` does the trick ?

Reply:
The paginator now can sort messages both ways - default is descending (newest first)